pkg/lang/csharp/asp_dotnet_core.go

```diff
@@ -0,0 +1,96 @@
+package csharp
+
+import (
+	"fmt"
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/filter"
+	sitter "github.com/smacker/go-tree-sitter"
+)
+
+type ASPDotNetCoreStartupClass struct {
+	Class                   TypeDeclaration
+	ConfigureMethod         MethodDeclaration
+	ConfigureServicesMethod MethodDeclaration
+}
+
+func FindASPDotnetCoreStartupClass(unit *core.ExecutionUnit) (*ASPDotNetCoreStartupClass, error) {
+	var startupClass *ASPDotNetCoreStartupClass
+	declarers := unit.GetDeclaringFiles()
+	if declarers == nil {
+		for _, csFile := range unit.FilesOfLang(CSharp) {
+			types := FindDeclarationsInFile[*TypeDeclaration](csFile).Declarations()
+			for _, t := range types {
+				if cls, found := getDotnetCoreStartupClass(t.Node); found {
+					if startupClass != nil {
+						return nil, fmt.Errorf("multiple ASP.NET Core startup classes found in execution unit [%s] <- [%s, %s]", unit.Name, startupClass.Class.Name, cls.Class.Name)
+					} else {
+						startupClass = &cls
+					}
+				}
+			}
+		}
+	}
+	for _, declarer := range declarers {
+		execUnitAnnotations := filter.NewSimpleFilter[*core.Annotation](func(a *core.Annotation) bool {
+			return a.Capability.Name == "execution_unit" &&
+				a.Capability.ID == unit.Name
+		}).Apply(declarer.Annotations().InSourceOrder()...)
+
+		if len(execUnitAnnotations) == 0 {
+			continue
+		}
+
+		for _, annotation := range execUnitAnnotations {
+			if cls, found := getDotnetCoreStartupClass(annotation.Node); found {
+				if startupClass != nil {
+					return nil, fmt.Errorf("multiple ASP.NET Core startup classes found in execution unit [%s] <- [%s, %s]", unit.Name, startupClass.Class.Name, cls.Class.Name)
+				} else {
+					startupClass = &cls
+				}
+			}
+		}
+	}
+	return startupClass, nil
+}
+
+func getDotnetCoreStartupClass(classNode *sitter.Node) (ASPDotNetCoreStartupClass, bool) {
+	classDeclaration, found := FindDeclarationAtNode[*TypeDeclaration](classNode)
+	if !found || classDeclaration.Visibility == VisibilityPrivate || classDeclaration.Kind != DeclarationKindClass {
+		return ASPDotNetCoreStartupClass{}, false
+	}
+	methods := FindDeclarationsAtNode[*MethodDeclaration](classNode).Declarations()
+	configureMethods := filter.NewSimpleFilter[*MethodDeclaration](func(md *MethodDeclaration) bool {
+		return md.Name == "Configure" &&
+			md.Visibility == VisibilityPublic &&
+			md.ReturnType == "void" &&
+			!md.IsStatic &&
+			!md.IsAbstract &&
+			len(md.Parameters) == 2 &&
+			IsValidTypeName(md.Parameters[0].TypeNode, "Microsoft.AspNetCore.Builder", "IApplicationBuilder") &&
+			IsValidTypeName(md.Parameters[1].TypeNode, "Microsoft.AspNetCore.Hosting", "IWebHostEnvironment")
```

Comment:
I don't see a much of a purpose in adding a one-off query when we can already pull the relevant details from declarations. Additionally, the type name validation for the parameters goes beyond what we can accomplish with pattern matching.

Reply:
The purpose would be removing the need to parse Visibility, ReturnType, IsStatic, IsAbstract entirely which would simplify a lot of the `declarations.go` stuff.

Reply:
Visibility has some implicit values that can't be captured in a query. I had considered making the rest of the modifiers retrievable on demand (like is currently done with attributes) rather than part of the model to simplify that aspect, but haven't prioritized it.

Reply:
something like this:  `func (d *TypeDeclaration) HasModifiers(modifier ...string) bool`